Fix mold instructions

Hofer-Julian opened the pull request:
- `bin` has to be appended to PATH not lib
- `lib` is not necessary at all
- Remove the `CARGO_HOME` part.
- Remove the last sentence. Let's just rewrite the instructions when gcc12.1 is out.


@SeaDve I tested it again, and as expected all you need to do is add `bin` to `PATH` to get clang.
Not sure what problems you encountered there.
Also why did you specify `CARGO_HOME` there?

CC: @bilelmoussaoui

SeaDve commented:
> Also why did you specify CARGO_HOME there?

Didn't work for me if it wasn't there.

But I'll try again in a bit, I might have missed something

SeaDve commented on README.md:
```diff
@@ -8,8 +8,8 @@ This extension contains various components of the [Rust](https://www.rust-lang.o
 In order to use the (fast) [mold](https://github.com/rui314/mold) linker:
 
 1. Add `org.freedesktop.Sdk.Extension.llvm13` allong with this extension in order to get clang.
```
I also had a typo here with `allong`

Hofer-Julian commented:
> > Also why did you specify CARGO_HOME there?
> 
> Didn't work for me if it wasn't there

Did you try adding `.cargo/config.toml` in the repo root?

Hofer-Julian committed:
Fix mold instructions

- `bin` has to be appended to PATH not lib
- `lib` is not necessary at all
- Remove the `CARGO_HOME` part.
- Remove the last sentence. Let's just rewrite the instructions when gcc12.1 is out
- Fix typo

SeaDve commented:
> > > Also why did you specify CARGO_HOME there?
> > 
> > 
> > Didn't work for me if it wasn't there
> 
> Did you try adding `.cargo/config.toml` in the repo root?

Oh repo root, not in `~/.cargo/config.toml`?

Hofer-Julian commented:
> > > > Also why did you specify CARGO_HOME there?
> > > 
> > > 
> > > Didn't work for me if it wasn't there
> > 
> > 
> > Did you try adding `.cargo/config.toml` in the repo root?
> 
> Oh repo root, not in `~/.cargo/config.toml`?

Yes, you only want mold to be used for this one project.

alatiera commented on README.md:
```diff
@@ -7,9 +7,9 @@ This extension contains various components of the [Rust](https://www.rust-lang.o
-3. Add `config.toml` with the following content in `CARGO_HOME` directory:
+1. Add `org.freedesktop.Sdk.Extension.llvm13` along with this extension in order to get clang.
+2. Add `/usr/lib/sdk/llvm13/bin` to `append-path`. See [llvm13 SDK extension readme](https://github.com/flathub/org.freedesktop.Sdk.Extension.llvm13) for more information.
+3. Add `.cargo/config.toml` in the root of the repository with the following content:
```
Ideally you want the config file to be only in your local `cargo_home` to avoid making everything go through mold regardless the environment

Hofer-Julian replied:
That's what happens if I add it at the repo root, no?

alatiera replied:
No, when you add it to the repo root, its treated as the configuration to be used for the whole repository, regardless if mold exists or not

alatiera replied:
Let's go with this for now

Erick555 commented:
You may also execute [source /usr/lib/sdk/llvm13/enable.sh](https://github.com/flathub/org.freedesktop.Sdk.Extension.llvm13#debuggingdevelopment) in order to get clang in `PATH`.

bilelmoussaoui commented:
> You may also execute [source /usr/lib/sdk/llvm13/enable.sh](https://github.com/flathub/org.freedesktop.Sdk.Extension.llvm13#debuggingdevelopment) in order to get clang in `PATH`.

That is only needed if you don't use append-path, isn't it?

Erick555 commented:
Yes, this is alternative to append-path. The advantage is it can be used on demand without specifying it in manifest first.

alatiera committed:
readme: Add debugging section

directly copied from the llvm extension

https://github.com/flathub/org.freedesktop.Sdk.Extension.llvm13
